src/IO/CompressionMethod.cpp

```diff
@@ -139,6 +149,12 @@ std::unique_ptr<WriteBuffer> wrapWriteBufferWithCompressionMethod(
     if (method == CompressionMethod::Bzip2)
         return std::make_unique<Bzip2WriteBuffer>(std::move(nested), level, buf_size, existing_memory, alignment);
 #endif
+
+#if USE_SNAPPY
+    if (method == CompressionMethod::Snappy)
+        return std::make_unique<SnappyWriteBuffer>(std::move(nested), buf_size, existing_memory, alignment);
```

Comment:
We can also pass and set compression level parameter as for other compression methods.